konveyor/pelorus
# Tekton compatibility for committime exporter

copied from #327

## Describe the behavior changes introduced in this PR

Adds the ability to get git commit information from Tekton runs.

It's a good enough implementation for "technology preview" status. There are open questions about how to extract the right information when there's near infinite flexibility in making a pipeline.

Right now, it tries to follow the (I believe) most common usage patterns:

- Looks up git information from the `git-checkout` `ClusterTask` that comes with Tekton
- Gets the image hash from the first `TaskRun` with a `taskResult` named `IMAGE_DIGEST`
- Gets the image location from the first `TaskRun` with `IMAGE` or `IMAGE_NAME` as a param, following patterns from various s2i and buildah tasks

I'm sure we'll need to figure out a way to add some flexibility. I've started exploring some possible solutions while experimenting with this, but that should come later.

Special thanks to @jlozano-rh for starting the work on this.

## Linked Issues?

resolves #224

## Testing Instructions

1. Fork _my_ fork of this repo and check out the tekton branch.
2. Set up the committime exporter. Make sure it is pointed to your fork. The deploytime exporter must also be there but can still point upstream.
3. Run the tekton demo with `./demo/demo-tekton $URL_FOR_FORK`.
4. Make sure the pipeline was successful and look at the sample page.
5. Use the prompt to make a change, and wait for it to show up.
6. Verify the metrics show up in the dashboard.

@redhat-cop/mdt

## Remaining Work

- [ ] feature-flag the tekton features
- [ ] resolve demo issue: occasional `unable to recognize no matches for kind "Pipeline" in version "tekton.dev/v1beta1"`
- [ ] demo should poll for the page to be ready
- [ ] git-clone task unavailable for the first run?
